cs-jClouds/src/main/java/io/cloudslang/content/jclouds/services/helpers/AWSSignatureHelper.java

```diff
@@ -0,0 +1,150 @@
+package io.cloudslang.content.jclouds.services.helpers;
+
+import io.cloudslang.content.jclouds.entities.constants.Constants;
+import org.apache.commons.lang3.StringUtils;
+
+import java.text.SimpleDateFormat;
+import java.util.*;
+
+/**
+ * Created by Mihai Tusa.
+ * 8/8/2016.
+ */
+public class AWSSignatureHelper {
+    private static final String DATE_FORMAT = "yyyyMMdd'T'HHmmss'Z'";
+    private static final String HYPHEN = "-";
+    private static final String SEMICOLON = ";";
+    private static final String TIME_ZONE = "UTC";
+
+    private static final char DOT_CHAR = '.';
+
+    /**
+     * The canonicalized (standardized) query string is formed by first sorting all the query
+     * parameters, then URI encoding both the key and value and then
+     * joining them, in order, separating key value pairs with an '&'.
+     *
+     * @param queryParameters The query parameters to be canonicalized.
+     * @return A canonicalized form for the specified query parameters.
+     */
+    public String canonicalizedQueryString(Map<String, String> queryParameters) {
+        List<Map.Entry<String, String>> sortedList = new ArrayList<>(queryParameters.entrySet());
+        Collections.sort(sortedList, new UTF8StringComparator());
+
+        StringBuilder queryString = new StringBuilder();
+        for (Map.Entry<String, String> entry : sortedList) {
+            queryString.append(entryToQuery(entry));
+        }
+        if (queryString.length() > 0) queryString.deleteCharAt(queryString.length() - 1);   //last &
+
+        return queryString.toString();
+    }
+
+    /**
+     * The canonicalized (standardized) headers string is formed by first sorting all the header
+     * parameters, then converting all header names to lowercase and
+     * trimming excess white space characters out of the header values
+     *
+     * @param headers The headers to be canonicalized.
+     * @return A canonicalized form for the specified headers.
+     */
+    public String canonicalizedHeadersString(Map<String, String> headers) {
+        List<Map.Entry<String, String>> sortedList = new ArrayList<>(headers.entrySet());
+        Collections.sort(sortedList, new Comparator<Map.Entry<String, String>>() {
+            public int compare(Map.Entry<String, String> o1, Map.Entry<String, String> o2) {
+                return (o1.getKey().compareToIgnoreCase(o2.getKey()));
+            }
+        });
```

Comment:
@gituser173 It's an ad-hoc overriding of Collections -> Comparator method compare(Object obj1, Object obj2) so I can use it to sort headers just before sign them.